gbpservice/nfp/orchestrator/config_drivers/heat_driver.py

```diff
@@ -787,6 +788,20 @@ def _update_node_config(self, auth_token, tenant_id, service_profile,
                  {'stack_data': stack_template, 'params': stack_params})
         return (stack_template, stack_params)
 
+    def parse_template_config_string(self, config_str):
+        service_config = tag_str = ''
+        for tag_str in [nfp_constants.HEAT_CONFIG_TAG,
+                        nfp_constants.CONFIG_INIT_TAG,
+                        nfp_constants.ANSIBLE_TAG]:
+            try:
+                service_config = config_str.split(tag_str + ':')[1]
+                break
+            except Exception:
```

Comment:
We should not use exception to proceed in positive cases. What if you get a different exception in which case we should not proceed. Rather check exact case and decide when to proceed and when not to.